modules/transactionpool/update.go

```diff
@@ -39,6 +46,46 @@ func (tp *TransactionPool) ProcessConsensusChange(cc modules.ConsensusChange) {
 	if err != nil {
 		tp.log.Println("ERROR: could not update the recent consensus change:", err)
 	}
+	err = tp.putBlockHeight(tp.dbTx, tp.blockHeight)
+	if err != nil {
+		tp.log.Println("ERROR: could not update the block height:", err)
+	}
+
+	// prune transactions older than maxTxnAge.
+	pruneTxns := make(map[types.TransactionID]struct{})
+	for txid, seenHeight := range tp.transactionHeights {
+		if tp.blockHeight-seenHeight > maxTxnAge {
+			delete(tp.transactionHeights, txid)
+			pruneTxns[txid] = struct{}{}
+		}
+	}
```

Comment:
hmm. I'm worried that scrolling through the entire list of transctions every block is not going to scale very well if we ever put the tpool at millions of transactions. (which just be a few hundred MBs).

I'm trying to think of more scalable ways to do this. Linear time stuff works up to thousands, but at tens of thousands you need to start reconsidering.